components/togglebox.tsx

```diff
@@ -0,0 +1,99 @@
+import { ChangeEvent, InputHTMLAttributes } from "react";
+import { AnimatePresence, motion } from "framer-motion";
+import { RemixiconComponentType } from "@remixicon/react";
+import { cn } from "./../lib/utils";
+import LoadingSpinner from "./loading-spinner";
+
+interface ToggleboxProps extends InputHTMLAttributes<HTMLInputElement> {
+  checked?: boolean;
+  onChange?: (e: ChangeEvent<HTMLInputElement>) => void;
+  icon?: RemixiconComponentType | null;
+  isLoading?: boolean;
+  name?: string;
+  label?: string;
+  description?: string;
+  containerClassName?: string;
+  titleClassName?: string;
+}
+
+export default function Togglebox({
+  name,
+  checked = false,
+  onChange,
+  icon: Icon = null,
+  isLoading = false,
+  label,
+  description,
+  containerClassName,
+  titleClassName,
+  ...props
+}: ToggleboxProps) {
+  const inputId = `togglebox-${name}-${props.value}`;
+
+  return (
+    <div className={cn("flex flex-row gap-2", containerClassName)}>
+      <AnimatePresence mode="wait" initial={false}>
+        {isLoading ? (
+          <motion.div
```

Comment:
now I wonder, when `isLoading`, I think to avoid flickering, how if we render the `LoadingSpinner` on the togglebox circle slider?

to demonstrate this, lets update the isloading story too, and sleep 1 second before setting the isloading to false, when sliding it...